src/mqtt.cpp

```diff
@@ -269,8 +302,8 @@ namespace mqtt {
         mqtt_client->subscribe(buf);
         sprintf(buf, "%s/down/#", config.mqttTopic);
         mqtt_client->subscribe(buf);
-        sprintf(buf, "%s/%u/up/status", config.mqttTopic, config.deviceId);
-        mqtt_client->publish(buf, "{\"online\":true}");
+        sprintf(statusmsg, "connected after %d attempts", connectionAttempts);
```

Comment:
Do you mind passing the reconnect count as a separate property instead of injecting it into the string?
I've got a `mqtt_client->publish(buf, "{\"online\":false}");` message in the (so far unpublished) branch for the portable monitor to signal the device going to disconnect before sleep.

Reply:
OK, see what you think about this reworked code... allows an arbitrary JSON doc to be provided as the status message (as a pointer, which the queue takes ownership of to free after serialization). 

This allows sending structured data (like offline:false, and connectionAttempts) but still makes it pretty simple to provide a nice wrapper for sending a simple message. 

Only 4 more bytes on the queue in the general case (e.g. other messages), but when providing a doc, the caller has control over how much memory to allocate.